src/lua/compat.lua

```diff
@@ -96,6 +103,12 @@ local options = {
         brief = SQL_SEQ_SCAN_DEFAULT_BRIEF,
         action = tweak_action('sql_seq_scan_default', true, false),
     },
+    box_tuple_new_arg_list = {
+        default = 'new',
+        obsolete = nil,
+        brief = BOX_TUPLE_NEW_ARG_LIST_BRIEF,
+        action = tweak_action('box_tuple_new_arg_list', true, false)
+    },
```

Comment:
I would think about a most obvious name for the compat option and the tweak.

To start the discussion:

| compat | tweak |
| ------ | ----- |
| `box_tuple_new_forbid_vararg` | `box_tuple_new_forbid_vararg` |

Reply:
I think the backward compatibility option should name the feature that is disabled.

Reply:
How `arg_list` characterizes the feature? Does it mean a list of fields in the second argument or unpacking the list into the multiple arguments?

Or it just means a function contract -- a kind of API version? If so, the 'list' word looks confusing.

Reply:
I see, how about `box_tuple_new_vararg`?

Reply:
Seems okay.

Reply:
Fixed.